scipy/_lib/_uarray/vectorcall.h

```diff
@@ -6,18 +6,16 @@ extern "C" {
 #endif
 
 // True if python supports vectorcall on custom classes
-#if (!defined(PYPY_VERSION) && (PY_VERSION_HEX >= 0x03080000))
+#if (!defined(PYPY_VERSION))
 #  define Q_HAS_VECTORCALL 1
 #else
 #  define Q_HAS_VECTORCALL 0
 #endif
 
 #if !Q_HAS_VECTORCALL
 #  define Q_Py_TPFLAGS_HAVE_VECTORCALL 0
-#elif (PY_VERSION_HEX >= 0x03090000)
-#  define Q_Py_TPFLAGS_HAVE_VECTORCALL Py_TPFLAGS_HAVE_VECTORCALL
 #else
-#  define Q_Py_TPFLAGS_HAVE_VECTORCALL _Py_TPFLAGS_HAVE_VECTORCALL
+#  define Q_Py_TPFLAGS_HAVE_VECTORCALL Py_TPFLAGS_HAVE_VECTORCALL
 #endif
```

Comment:
Actually, now that there's no version distinction here anymore, we can completely remove `Q_HAS_VECTORCALL` and just merge that into one `#if / #else`